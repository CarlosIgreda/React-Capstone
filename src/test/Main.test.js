import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import CoinsList from '../components/Main';

const mockStore = configureStore([thunk]);

test('renders coins list correctly', () => {
  const store = mockStore({
    coins: {
      currency: {
        BTC: {
          symbol: 'BTC',
          icon_url: 'btc-icon-url',
          name: 'Bitcoin',
          current_price: 50000,
        },
      },
      loading: false,
      error: null,
    },
  });

  const { container } = render(
    <Router>
      <Provider store={store}>
        <CoinsList />
      </Provider>
    </Router>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
