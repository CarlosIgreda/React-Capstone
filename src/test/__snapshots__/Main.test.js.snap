// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders coins list correctly 1`] = `
<nav
  class="body-tertiary nav-container"
>
  <div
    class="container-fluid search-container"
  >
    <form
      class="d-flex nav-form"
      role="search"
    >
      <input
        aria-label="Search"
        class="form-control me-2"
        placeholder="Search your coin"
        type="search"
        value=""
      />
      <button
        class="btn btn-outline-success button-search"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass "
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
            fill="currentColor"
          />
        </svg>
      </button>
    </form>
  </div>
</nav>
`;
